VikingQuiz.UI/src/Components/QuizComponent/StartQuizComponent/StartQuizComponent.tsx

```diff
@@ -0,0 +1,273 @@
+import SubmitButton from '../../Buttons/SubmitButton/SubmitButton';
+import UploadButton from '../../Buttons/UploadButton/UploadButton'
+import HttpService from '../../../services/HttpService';
+
+interface IState {
+    title: string,
+    selectedFile: any,
+    imageError: string,
+    titleError: string,
+    isValid: boolean,
+    saved: boolean
+}
+
+interface IProps{
+    editMode: boolean
+    quizId: number,
+    save: any
+}
+
+const initialTitle: string = 'add quiz title';
+
+class StartQuizComponent extends React.Component<IProps, IState>{
+
+    public fileInputRef: any = React.createRef();
+    public previewImageRef: any = React.createRef();
+    public httpService: any = new HttpService();
+
+    public state = {
+        title: initialTitle,
+        selectedFile: '',
+        imageError: '',
+        titleError: '',
+        isValid: false,
+        saved: false
+    }
+
+    public componentWillMount(){
+        if(this.props.editMode){
+            const getQuizUrl: string = 'http://localhost:60151/api/quizzes/' + this.props.quizId; 
+            this.httpService.getWithToken(getQuizUrl)
+            .then((response: any) => {
+                // this part gets the photo at the specified url
+                const imageUrl: string = response.data.pictureUrl;
+                const titleValue: string = response.data.title;
+
+                console.log(imageUrl, titleValue);
+                this.getImageBlobFromURL(imageUrl, titleValue);
+            })
+            .catch((error: any) => console.log(error));
+        }
+    }
+
+    public changeTitleHandler = (event: any) => {
+        const currentTitle: string = event.target.value
+        this.isTitleValid(currentTitle);
+        this.setState({
+            title: event.target.value
+        });
+    }
+
+    public focusTitleHandler = () => {
+        if(this.state.title === initialTitle){
+            this.setState({
+                title: ''
+            })
+            this.isTitleValid('');
+        }
+    }
+
+    public fileSelectHandler = (event: any) => {
+        const file = event.target.files[0];
+        if(!this.isImageValid(file)){
+            return;
+        }
+        const reader = new FileReader();
+        
+        reader.onload = (e: any) => {
+            this.previewImageRef.current.src = e.target.result;
+        }
+        
+        reader.readAsDataURL(file);
+        this.setState({
+            selectedFile: file
+        });
+    }
+
+    public uploadPhotoHandler = () => {
+        this.fileInputRef.current.click();
+    }
+
+    public saveQuizHandler = () => {
+
+        if(!this.allValid()){
+            return;
+        }
+        
+        const fd = new FormData();
+        fd.append('files', this.state.selectedFile);
+        fd.append('title', this.state.title.toUpperCase());
+
+        let url: string = '';
+        if(this.props.editMode){
+            url = 'http://localhost:60151/api/quizzes/' + this.props.quizId;
```

Comment:
If you would have stored the url in a constant, it would have been way easier to change it in all places at once.